x-pack/plugins/stack_connectors/server/connector_types/inference/inference.ts

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+
+import { PassThrough, Stream } from 'stream';
+import { IncomingMessage } from 'http';
+
+import { AxiosError } from 'axios';
+import {
+  InferenceInferenceRequest,
+  InferenceInferenceResponse,
+  InferenceTaskType,
+} from '@elastic/elasticsearch/lib/api/types';
+import {
+  ChatCompleteParamsSchema,
+  RerankParamsSchema,
+  SparseEmbeddingParamsSchema,
+  TextEmbeddingParamsSchema,
+} from '../../../common/inference/schema';
+import {
+  Config,
+  Secrets,
+  ChatCompleteParams,
+  ChatCompleteResponse,
+  StreamingResponse,
+  RerankParams,
+  RerankResponse,
+  SparseEmbeddingParams,
+  SparseEmbeddingResponse,
+  TextEmbeddingParams,
+  TextEmbeddingResponse,
+} from '../../../common/inference/types';
+import { SUB_ACTION } from '../../../common/inference/constants';
+
+export class InferenceConnector extends SubActionConnector<Config, Secrets> {
+  // Not using Axios
+  protected getResponseErrorMessage(error: AxiosError): string {
+    throw new Error('Method not implemented.');
+  }
+
+  private inferenceId;
+  private taskType;
+
+  constructor(params: ServiceParams<Config, Secrets>) {
+    super(params);
+
+    this.provider = this.config.provider;
+    this.taskType = this.config.taskType;
+    this.inferenceId = this.config.inferenceId;
+    this.logger = this.logger;
+    this.connectorID = this.connector.id;
+    this.connectorTokenClient = params.services.connectorTokenClient;
+
+    this.registerSubActions();
+  }
+
+  private registerSubActions() {
+    this.registerSubAction({
+      name: SUB_ACTION.COMPLETION,
+      method: 'performApiCompletion',
+      schema: ChatCompleteParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.RERANK,
+      method: 'performApiRerank',
+      schema: RerankParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.SPARSE_EMBEDDING,
+      method: 'performApiSparseEmbedding',
+      schema: SparseEmbeddingParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.TEXT_EMBEDDING,
+      method: 'performApiTextEmbedding',
+      schema: TextEmbeddingParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.COMPLETION_STREAM,
+      method: 'performApiCompletionStream',
+      schema: ChatCompleteParamsSchema,
+    });
+  }
+
+  /**
+   * responsible for making a POST request to the Inference API chat completetion task endpoint and returning the response data
+   * @param body The stringified request body to be sent in the POST request.
```

Comment:
Could you please update the comments on these methods to accurately describe the input? it seems here from your tests the input should just be the user's string input, not a request body